pegasus/sites.v3/code.org/public/ops-dashboard/index.haml

```diff
@@ -501,15 +502,22 @@ title: "Ops Dashboard"
         if (teacher.id) {
           teacher.editing = false;
         } else {
-          $scope.cohort.teachers.splice($scope.cohort.teachers.indexOf(teacher), 1); // remove from array
+          if (teacher.newTeacher) {
```

Comment:
I think you can just check for the absence of teacher.id instead of creating this new property